services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreBackupVersionCleanupService.java

```diff
@@ -276,6 +276,16 @@ protected boolean cleanupBackupVersion(Store store, String clusterName) {
                 : v.getNumber() < currentVersion)
         .collect(Collectors.toList());
 
+    // If there are still leaking versions due to consecutive repushes, there could be versions with repushSourceVersion
+    // does not match current version, delete them after backup retention period.
+    if (readyToBeRemovedVersions.isEmpty()) {
+      for (Version version: versions) {
+        if (version.getNumber() < currentVersion && store.getLatestVersionPromoteToCurrentTimestamp()
```

Comment:
My understanding is that this is the same as removing the above filtering?

Reply:
As discussed offline, there is a subtle difference if we remove the above filtering which would lead to leaking 2 backup version for the whole retention period.